projects/fusion-ui/components/table/common/services/table.service.ts

```diff
@@ -248,7 +249,54 @@ export class TableService {
     }
 
     isRowReadOnly(row: any): boolean {
-        return !!row.rowMetaData && !!row.rowMetaData.readonly;
+        return !!row.rowMetaData?.readonly;
+    }
+
+    isRowInRequest(row: any): boolean {
+        return row.rowMetaData?.inRequest;
+    }
+
+    toggleRowInRequest(row: any, isInRequest) {
+        this.insureRowMetaData(row);
+        row.rowMetaData.inRequest = isInRequest;
+    }
+
+    setRowspanColumnsData(rows: any[], columnsKeys: string[]) {
+        rows.forEach(row => {
+            this.insureRowMetaData(row);
+            if (Object.values(row).some(val => Array.isArray(val))) {
+                // set rowspan row metadata
+                this.hasRowspanRows = true;
+                row.rowMetaData.rowspanColumnsData = this.getRowspanColumns(row, columnsKeys);
+                row.rowMetaData.maxRowspanInColumn = Math.max(...(Object.values(row.rowMetaData.rowspanColumnsData) as number[]));
+            }
+        });
+    }
+
+    getRowspanColumnsData(row: any): {[key: string]: number} {
+        return row.rowMetaData?.rowspanColumnsData;
```

Comment:
where do u use this ??

Reply:
In template - see https://github.com/ironSource/fusion-ui/pull/164#discussion_r1098577516